src/resources/assets/js/geo/components/filterByLocationComponent.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+    <div class="filter-select">
+        <geo-map-modal id="gmm" :text="text" :trigger="trigger" :volumeId="volumeId" v-on:on="submit"></geo-map-modal>
+        <button type="submit" class="btn btn-default pull-right position" @click="trigger = !trigger">Add rule</button>
+    </div>
+</template>
+
+<script>
+import GeoMapModal from './geoMapModal.vue';
+
+// create custom event to update rule upon changes in sessionStorage
+const customEvent = new Event('storageUpdate', {
+    bubbles: true,
+    cancelable: true,
+    composed: false
+  });
```

Comment:
Yes, I agree, appending another "filter by location" rule feels like the most intuitive way! But as I understood it, we agreed on "replacing" the rule if there is already a "filter by location" rule in the list. This is because one cannot differentiate the individual "filter by location" rules from each other (there is no hint like labelname or username as in the other selection-filters).

Reply:
Regarding the second point: I get the problem with global events like that. I did it this way because I could not really figure out how to pass the event in a "Vue way" from the filterByLocationComponent to the listComponent of the [geoFilter.js](https://github.com/biigle/geo/blob/301f08a24f472cb70218807ff8fc9d194fc33f83/src/resources/assets/js/volumes/geoFilter.js#L14C9-L14C22). This is needed, because only the listComponent of the filter is able to call the refreshRule-function (see [link](https://github.com/biigle/core/blob/169200ebe4b9b38ce2bda7e64f90c24af93e8e4e/resources/views/volumes/show/filters.blade.php#L39)). If you have a suggestion for me here, I would try to implement it differently.